app/db/connection_manager.py

```
@@ -0,0 +1,443 @@
+            yield conn
+            
+        except Exception as e:
+            self._connection_stats["failed_connections"] += 1
+            self._connection_stats["last_error"] = str(e)
+            logger.error(f"Database connection error: {e}")
+            
+            if conn:
+                try:
+                    conn.rollback()
+                except:
+                    pass
```

Ruff (0.11.9)

Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(Exception)`

(SIM105)